src/+soilmoisture/calculateBoundaryConditions.m

```diff
@@ -72,7 +72,7 @@
         % Check applied infiltration doesn't exceed infiltration capacity
         topThick = 5; % first 5 cm of the soil
         satCap = SoilProperties.theta_s0 * topThick; % saturation capacity represented by saturated water content of the top 5 cm of the soil
-        actTheta = ModelSettings.DeltZ(51:54) * SoilVariables.Theta_UU(51:54, 1); % actual moisture of the top 5 cm of the soil
+        actTheta = ModelSettings.DeltZ(end - 4:end - 1) * SoilVariables.Theta_UU(end - 4:end - 1, 1); % actual moisture of the top 5 cm of the soil
```

Comment:
Here you are assuming that the lengths of all changed variables are 55. But the length of `ModelSettings.DeltZ` is 54 similar to `ModelSettings.NL`. So,` end - 4:end - 1` is not the same as `51:54` for `ModelSettings.DeltZ`.

Reply:
Thanks for catching this mistake. you are right, `ModelSettings.DeltZ` length is 54 while `SoilVariables.Theta_UU` is 55. so the line should be corrected to

`actTheta = ModelSettings.DeltZ(end - 3:end) * SoilVariables.Theta_UU(end - 4:end - 1, 1);`